### PR TITLE
add openai support

### PR DESCRIPTION
**Pull request by hoang-shakudo**
Add openai-support for graph-rag

### CODE REVIEW

File: graph-rag/app.py [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
-from neo4j import GraphDatabase, AsyncGraphDatabase
+from neo4j import GraphDatabase
 from langchain_community.embeddings import OllamaEmbeddings
 from langchain_community.chat_models import ChatOllama
+from langchain_community.embeddings
 
-from prompts import PROMPT_QWEN
+from langchain_community.embeddings.openai import OpenAIEmbeddings
+from langchain_community.chat_models.openai import ChatOpenAI
+os.environ['OPENAI_API_KEY'] = '***'
```
Comment (chyuen-devsentient) [inline graph-rag/app.py:11]: add a comment to replace this

Comment (chyuen-devsentient) [reply graph-rag/app.py:11]: also, given this is a public doc, the imports and variables are all over the place. Can we follow some best practices for organizing the files?

File: graph-rag/app.py [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+from langchain_community.chat_models.ollama import ChatOllama
+from langchain_community.embeddings.openai import OpenAIEmbeddings
+from langchain_community.chat_models.openai import ChatOpenAI
+from prompts import PROMPT_QWEN, PROMPT_OPENAI
 
 app = FastAPI()
 
+### For OpenAI users, please uncomment these following lines:
+# use_openai = os.environ.get('OPENAI_API_KEY', None) is not None
+# if use_openai:
```
Comment (chyuen-devsentient) [inline graph-rag/app.py:14]: openAI users have to uncomment this, but the value of `use_openai` is still not set -- should we remove the "if"

Comment (hoang-shakudo) [reply graph-rag/app.py:14]: `use_openai = os.environ.get('OPENAI_API_KEY', None) is not None` I think this is how I set the value for `use_openai`

**Review by chyuen-devsentient — APPROVED** [review]

**Review by chyuen-devsentient — COMMENTED** [review]
sorry last comment -- can we move the neo4j, ollama , openAI setup to a different file? It's currently redeclared in 3 different files

Also some imports in ingest.py are still messy

**Review by chyuen-devsentient — APPROVED** [review]

**Pull request merged**